aws/resource_aws_pinpoint_app.go

```diff
@@ -0,0 +1,435 @@
+
+	return nil
+}
+
+func resourceAwsPinpointAppDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).pinpointconn
+
+	log.Printf("[DEBUG] Pinpoint Delete App: %s", d.Id())
+	_, err := conn.DeleteApp(&pinpoint.DeleteAppInput{
+		ApplicationId: aws.String(d.Id()),
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func expandCampaignHook(d *schema.ResourceData) *pinpoint.CampaignHook {
+	configs := d.Get("campaign_hook").(*schema.Set).List()
+	if configs == nil || len(configs) == 0 {
+		return nil
+	}
+
+	m := configs[0].(map[string]interface{})
+
+	ch := &pinpoint.CampaignHook{}
+
+	if m["lambda_function_name"] != nil {
+		ch.LambdaFunctionName = aws.String(m["lambda_function_name"].(string))
+	}
+
+	if m["mode"] != nil {
+		ch.Mode = aws.String(m["mode"].(string))
+	}
+
+	if m["web_url"] != nil {
+		ch.WebUrl = aws.String(m["web_url"].(string))
+	}
+
+	return ch
+}
+
+func flattenCampaignHook(ch *pinpoint.CampaignHook) []interface{} {
+	l := make([]interface{}, 0)
+
+	m := map[string]interface{}{}
+
+	if ch.LambdaFunctionName != nil {
+		m["lambda_function_name"] = *ch.LambdaFunctionName
+	}
+
+	if ch.Mode != nil && *ch.Mode != "" {
+		m["mode"] = *ch.Mode
+	}
+
+	if ch.WebUrl != nil {
+		m["web_url"] = *ch.WebUrl
+	}
+
+	if len(m) <= 0 {
+		return nil
+	}
+
+	l = append(l, m)
+
+	return l
+}
+
+func expandCampaignLimits(s *schema.Set) *pinpoint.CampaignLimits {
+	if s == nil || s.Len() == 0 {
+		return nil
+	}
+	m := s.List()[0].(map[string]interface{})
+
+	cl := pinpoint.CampaignLimits{}
+
+	if m["daily"] != nil {
+		cl.Daily = aws.Int64(int64(m["daily"].(int)))
+	}
+
+	if m["maximum_duration"] != nil {
+		cl.MaximumDuration = aws.Int64(int64(m["maximum_duration"].(int)))
+	}
+
+	if m["messages_per_second"] != nil {
+		cl.MessagesPerSecond = aws.Int64(int64(m["messages_per_second"].(int)))
+	}
+	if m["total"] != nil {
+		cl.Total = aws.Int64(int64(m["total"].(int)))
+	}
+
+	return &cl
+}
+
+func flattenCampaignLimits(cl *pinpoint.CampaignLimits) []interface{} {
+	l := make([]interface{}, 0)
+
+	m := map[string]interface{}{}
+
+	if cl.Daily != nil {
+		m["daily"] = *cl.Daily
+	}
+	if cl.MaximumDuration != nil {
+		m["maximum_duration"] = *cl.MaximumDuration
+	}
+	if cl.MessagesPerSecond != nil {
+		m["messages_per_second"] = *cl.MessagesPerSecond
+	}
+
+	if cl.Total != nil {
+		m["total"] = *cl.Total
+	}
+
+	if len(m) <= 0 {
+		return nil
+	}
+
+	l = append(l, m)
+
+	return l
+}
+
+func expandQuietTime(s *schema.Set) *pinpoint.QuietTime {
+	if s == nil || s.Len() == 0 {
+		return nil
+	}
+
+	m := s.List()[0].(map[string]interface{})
+
+	qt := pinpoint.QuietTime{}
+
+	if m["end"] != nil {
+		qt.End = aws.String(m["end"].(string))
+	}
+
+	if m["start"] != nil {
+		qt.Start = aws.String(m["start"].(string))
+	}
+
+	return &qt
+}
+
+func flattenQuietTime(qt *pinpoint.QuietTime) []interface{} {
+	l := make([]interface{}, 0)
+
+	m := map[string]interface{}{}
+
+	if qt.End != nil {
+		m["end"] = qt.End
```

Comment:
I believe this would be found by the `d.Set()` error checking, but this should be converting the string pointer to a string:

```go
m["end"] = aws.StringValue(qt.End)
```

Reply:
Fixed in https://github.com/terraform-providers/terraform-provider-aws/commit/ea1e963c22edd9cc3720899eecedc4bd13f36c6d